### PR TITLE
Add shop_cookie_policy Option on Connect

### PR DESCRIPTION
**Pull request by mrwhizzy**

### CODE REVIEW

File: spec/session_spec.coffee [hunk 1]
```diff
@@ -121,14 +121,15 @@ on_create_session_retrieval_tests = (cookie_exists = false, session_from_cookie
...
-    @cookie_policy     = 'full'
-    @cookie_policy_prm = 'f'
-    @flavor            = 'flavor'
-    @metadata          = { app_type: 'web', cp: @cookie_policy_prm, tags: 'tag1,tag2' }
+    @type_create        = 'create'
+    @type_connect       = 'connect'
+    @shop_code          = 'shop_code_1'
+    @analytics_session  = 'dummy_analytics_session_hash'
+    @cookie_policy      = 'full'
+    @shop_cookie_policy = 'full'
```
Comment (MrGoumX) [inline spec/session_spec.coffee:129]: We can revert this change it's not needed

File: spec/session_spec.coffee [hunk 2]
```diff
@@ -508,6 +509,32 @@ describe 'Session', ->
 
             expect(@biskoto.get(@basic_meta_cookie_name)).to.deep.equal metadata
 
+        context 'and shop_cookie_policy is specified as basic', ->
+          beforeEach ->
+            @shop_cookie_policy = 'basic'
```
Comment (MrGoumX) [inline spec/session_spec.coffee:514]: This line does not effect the run of the spec because init is called a little bit differently

File: spec/session_spec.coffee [hunk 3]
```diff
@@ -508,6 +509,32 @@ describe 'Session', ->
...
+
+          it 'sets the session first-party basic cookie value', ->
+            metadata = { app_type: @metadata.app_type, cp: 'b', tags: @metadata.tags }
+            @stub_params({ session: @analytics_session, metadata: metadata })
+
+            @init()
```
Comment (MrGoumX) [inline spec/session_spec.coffee:520]: Here we need to pass an optional parameter `shop_cookie_policy` and by default this needs to be `full`. The init method is defined in L299

File: spec/session_spec.coffee [hunk 4]
```diff
@@ -508,6 +509,32 @@ describe 'Session', ->
...
+
+            expect(@biskoto.get(@basic_meta_cookie_name)).to.deep.equal metadata
+
+          it 'does not sets the session first-party full cookie value', ->
+            expect(@biskoto.get(@session_cookie_name)).to.be.null
+
+          it 'does not sets the metadata first-party full cookie value', ->
+            expect(@biskoto.get(@meta_cookie_name)).to.be.null
```
Comment (MrGoumX) [inline spec/session_spec.coffee:536]: The init method is not called and that's why the specs pass. If we call init the specs fail.

Comment (MrGoumX) [reply spec/session_spec.coffee:536]: Also if we call the init method the metadata cookie policy must be full

File: src/session.coffee [hunk 5]
```diff
@@ -26,18 +27,22 @@ define [
...
 
-        connect: (shop_code)->
+        connect: (shop_code, shop_cookie_policy = 'full')->
           return console?.warn?('Connect called without a shop code') unless shop_code
 
           # connect should be called only once
           return console?.warn?('Connect called multiple times') if @shop_code
 
           @shop_code = shop_code
+          @shop_cookie_policy = if shop_cookie_policy in SHOP_COOKIE_OPTIONS then shop_cookie_policy else 'full'
```
Comment (vinPopulaire) [inline src/session.coffee:41]: in order to avoid this, we could replace any `== 'full'` to `!= 'basic'`

Comment (alexunder193) [reply src/session.coffee:41]: It was the implementation I did in the past fixups. We did it with the constant in order to understand that these 2 are the acceptable cases.

Comment (alexunder193) [reply src/session.coffee:41]: It's easier for someone to understand the possible cases

**Comment by vinPopulaire** [comment]
In `actions_manager.coffee.rb:47` we have a condition where we don't show the OrderStash widget to users that have not accepted the full cookies from the skroutz site if I'm not mistaken. Do we want to add the check for the shop's cookies as well?

File: src/session.coffee [hunk 6]
```diff
@@ -165,20 +170,43 @@ define [
...
+      if @metadata
+        options = { domain: @domain } if @domain
+
+        # Delete full cookie if user changed preferences
+        full_metadata_cookie = Biskoto.get Settings.cookies.full.meta.name
+        Biskoto.expire Settings.cookies.full.meta.name, options if full_metadata_cookie
+
+      full_cache_options = expires: Settings.cookies.full.analytics.duration
+      full_cache_options.domain = @domain if @domain
+      Biskoto.expire Settings.cookies.full.analytics.name, full_cache_options
```
Comment (vinPopulaire) [inline src/session.coffee:210]: Is there a reason why we don't check here if a `Biskoto.get(Settings.cookies.full.analytics.name)` exists, as we do in the previous cookies?

File: spec/session_spec.coffee [hunk 7]
```diff
@@ -508,6 +508,181 @@ describe 'Session', ->
...
+
+            it 'sets the metadata first-party basic cookie value', ->
+              metadata = { app_type: @metadata.app_type, cp: 'f', tags: @metadata.tags }
+              @stub_params({ session: @analytics_session, metadata: metadata })
+
+              @init('kokolala')
+
+              expect(@biskoto.get(@basic_meta_cookie_name)).to.deep.equal metadata
+
+            it 'does not sets the session first-party full cookie value', ->
```
Comment (vinPopulaire) [inline spec/session_spec.coffee:578]: rename, we want to set the full cookies

File: spec/session_spec.coffee [hunk 8]
```diff
@@ -508,6 +508,181 @@ describe 'Session', ->
...
+              metadata = { app_type: @metadata.app_type, cp: 'f', tags: @metadata.tags }
+              @stub_params({ session: @analytics_session, metadata: metadata })
+
+              @init('kokolala')
+
+              expect(@biskoto.get(@session_cookie_name).session).to.equal @analytics_session
+
+            it 'does not sets the metadata first-party full cookie value', ->
```
Comment (vinPopulaire) [inline spec/session_spec.coffee:586]: same

Comment (alexunder193) [reply spec/session_spec.coffee:586]: yes!

File: spec/session_spec.coffee (hunk 7 above)
Comment (alexunder193) [reply spec/session_spec.coffee:578]: yes!

**Review by alexunder193 — APPROVED** [review]
LGTM +1
